packages/jupyter-ai/jupyter_ai/chat_handlers/export.py

```diff
@@ -25,21 +29,18 @@ def chat_message_to_markdown(self, message):
         else:
             return ""
 
-    # Multiple chat histories in separate files
-    def get_chat_filename(self, path="./chat_history.md"):
-        filename, extension = os.path.splitext(path)
-        counter = 1
-        while os.path.exists(path):
-            path = filename + "_" + str(counter) + ".md"
-            counter += 1
-        return path
-
-    async def process_message(self, _):
+    # Write the chat history to a markdown file with a timestamp
+    async def process_message(self, message: HumanChatMessage):
         markdown_content = "\n\n".join(
             self.chat_message_to_markdown(msg) for msg in self._chat_history
         )
-        # Write the markdown content to a file or do whatever you want with it
-        chat_filename = self.get_chat_filename()
+        args = self.parse_args(message)
+        if len(args.path) > 0:
+            file_id = args.path[0]
+        else:
+            file_id = "chat_history"
+        outfile = f"{file_id}-{datetime.now():%Y-%m-%d-%H-%M}.md"
+        chat_filename = os.path.join(self.root_dir, outfile)
         with open(chat_filename, "w") as chat_history:
```

Comment:
Looks good, a few minor suggestions to make this more readable. 
1. Can use the shorthand for the if condition, is more readable in this case.
2. Could use better names than file_id, and outfile. Suggesting using `chat_filename` and `chat_file` here.

```suggestion
        chat_filename = args.path[0] if args.path else "chat_history"
        chat_filename = f"{chat_filename}-{datetime.now():%Y-%m-%d-%H-%M}.md"
        chat_file = os.path.join(self.root_dir, chat_filename)
        with open(chat_file, "w") as chat_history:
```

Reply:
@3coins Great suggestions! All implemented.